### DISCUSSION TITLE
Password owner HA

### DISCUSSION THREAD

**Discussion by MapoCaos** [indomus/forum#2822] [HUB: Home Assistant]
Ciao,
ho appena installato HA supervised su mini pc apposito, sono alle prime armi su linux ma ho un problema di password dell'account owner. Ho fatto accesso su pc e telefono, dopo qualche giorno ho provato a fare l'accesso dinuovo  e non riesco più, mi dice utente o pass sbagliati. Io mi sono segnato la pass (mi sembra strano) ma forse ho sbagliato a segnarla. Adesso non riesco più ad entrare, ho solo più il cell collegato. Ho provato a seguire la guida ufficiale: https://www.home-assistant.io/docs/locked_out/#forgot-password però da app non riesco a reinpostarla e nella sezione: "To reset an owner’s password, via console" della guida non riesco a farla, in particolare non riesco a capire: "You can access the [Home Assistant console](https://www.home-assistant.io/hassio/commandline/) on the device itself (not via the SSH terminal from the add-ons). " come faccio accedere alla console HA? accedendo dal pc su cui è installato non so come fare e quando provo anche da app dal terminale di mi dice : `zsh: command not found: auth`  che significa sempre che non sono su HA terminale. Ora da super ignorante ho provato anche a disinstallare  HA supervised per partire da zero (non avevo ancora fatto nulla) ma non sono riuscito neanche a fare quello. Ora l'ultima spiaggia da niubbo è reinstallare debian 12 e HA... qualcuno saprebbe aiutarmi? grazie

**Comment by Scialla** [comment]
le istruzioni per risolvere sono quelle che hai provato a seguire  ma, come avrai notato, sono diverse a seconda del tipo di installazione che hai fatto e che non hai specificato qui

**Reply by MapoCaos** [reply]
Ciao grazie per la risposta. 
Scusa ho cercato di scrivere più info possibile ma ho dimenticato di scrivere quello. Ho installato HA supervised su docker su mini pc con debian 12. La scelta è dovuta al fatto che non conosco linux (e purtroppo non ho molto tempo da dedicare ad impararlo bene) e quindi HA core avrei avuto più difficoltà ma volevo cmq avere la possibilità di far girare altri programmi come ad esempio Pi-hole (cosa che se nn sbaglio HA OS, non mi avrebbe permesso).
Altra cosa: di solito accedo a linux tramite ssh dal portatile (xk il mini pc è vicino al router) però forse non è possibile fare queste impostazione da remoto quindi lo stacco e lo attacco alla tv. Mentre VNC non c'è stato verso di farlo funzionare, neanche con diverse guide su internet.

**Reply by Scialla** [reply]
quindi, seguendo la doc ufficiale, hai fatto:
TO RESET A USER’S PASSWORD, VIA THE CONTAINER COMMAND LINE
If you are running Home Assistant in a container, you can use the command line in the container with the hass command to change your password. The steps below refer to a Home Assistant container in Docker named homeassistant. Note that while working in the container, commands will take a few moments to execute.

`docker exec -it homeassistant bash` to open to the container command line
`hass` to create a default user, if this is your first time using the tool
`hass --script auth --config /config change_password existing_user new_password` to change the password
`exit` to exit the container command line
`docker restart homeassistant` to restart the container.

ovviamente dalla cartella ove è presente il tuo file docker-compose.yaml o dove è presente il container di homeassistant, considerando che si chiami home assistante oppure modificandolo col nome che gli hai dato (es. home-assistant)

nel caso avessi usato `docker compose` il primo comando diventa `docker compose exec -it homeassistant bash`

**Reply by MapoCaos** [reply]
In realtà io ho seguito: "TO RESET AN OWNER’S PASSWORD, VIA CONSOLE" però seguendo questo che mi hai citato mi dice cosi adesso:
`deb@debian:~$ docker exec -it homeassistant bash`
`homeassistant:/config# hass`
`2024-01-08 20:08:47.281 ERROR (MainThread) [homeassistant.componentsd to create HTTP server at port 8123: [Errno 98] error while attempton address ('::', 8123, 0, 0): address in use`
poi mi sembra che si blocchi perchè non risponde ai comandi
`hass --script auth --config /config change_password Edoardo prova123`
`exit`
`docker restart homeassistant`

la prima volta mi ha detto `User not exist` (o qualcosa del genere)

Però io non sono owner? cioè il primo account che ho creato

**Reply by Scialla** [reply]
prova prima a stoppare il container e poi a ripetere i comandi
`docker stop homeassistant`

**Reply by snakuzzo** [reply]
Una volta che sei nella Shell del container non devi digitare `hass` e basta...se lo lanci  viene avviato homeassistant ed è normale che ti dia quell'errore...visto che è già up&running

**Reply by Scialla** [reply]
però scusa, se al comando `hass --script auth --config /config change_password Edoardo prova123` ti ha restituito `user not exist` ti sei dimenticato il nome del tuo user.... visto che dici di essere ancora loggato sull'app del telefono, vatti a riguardare il nome del tuo user

**Reply by snakuzzo** [reply]
E comunque se sei già loggato con il tuo utente (che sarà sicuramente amministratore), puoi creare un altro utente e forzare la password al tuo ...

https://www.home-assistant.io/docs/locked_out/#to-reset-a-password-while-still-logged-in-including-supervised

**Reply by snakuzzo** [reply]
Considerando che il mio utente sia `prova`

Puoi farlo da command line accedendo alla console di HA...

![image](https://github.com/indomus/forum/assets/7205371/8532dfe8-2cbc-4edf-94c8-1d7347eb6421)

**Reply by snakuzzo** [reply]
Se invece lo fai collegandoti in SSH (incluso addon) non funziona (è spiegato nella documentazione):

![image](https://github.com/indomus/forum/assets/7205371/8d4d1165-f5c6-4711-a6bb-393f06ff87a4)

**Reply by snakuzzo** [reply]
quindi...o fai come ti ho detto prima creando un altro utente, oppure accedendo al container...

![image](https://github.com/indomus/forum/assets/7205371/6841e307-ef5d-406e-9148-e60f5576295d)

...e poi riavviando Core (non serve riavviare l'intero sistema)

![image](https://github.com/indomus/forum/assets/7205371/5a297615-d417-4150-a896-3c5795ede98f)

**Reply by snakuzzo** [reply]
magari il suo nome visualizzato è `Eduardo`, ma il nome utente è `eduardo`

**Reply by snakuzzo** [reply]
Se non ricordi il nome utente, per sicurezza cercalo nel file `auth`

![image](https://github.com/indomus/forum/assets/7205371/6ed6568e-e8c7-4eaa-a6be-fb514448f40f)

**Reply by MapoCaos** [reply]
Ciao, grazie mille per le risposte e scusate per il ritardo nella mia ma non ho potuto fare prima.
Ho provato a verificare prima il nome utente come mi avete consigliato ed era quello che è errato, grazie mille perchè non sapevo come verificarlo. Alla fine ho dovuto inserire la password nuova che avevo modificato da app, ma con cui non riuscivo ad entrare (grazie.. era l'utente errato!). Grazie mille dei suggerimenti che mi avete dato, me li segno per le prossime volte!
L'unica cosa che vorrei capire è: perchè quando inserisco il comando `docker exec -it homeassistant bash` non mi viene la scritta HOME ASSISTANT grande come negli screen di Snakuzzo? a me compare solo `homeassistant:/config#` sia da SSH che da console "terminale"

Un ultima domanda se posso: io sono riuscito ad accedere da 192.168.178.xxx:8123 mentre non sono riuscito da homeassistant.local:8123 quindi sul mio fritzbox 7530 devo attivare il multicast DNS (mDNS), ma non ho trovato come fare e soprattuto, è consigliabile farlo?

Grazie mille ancora per il supporto

**Reply by snakuzzo** [reply]
Riguarda bene gli screenshot. Io le prove le ho fatte da HA OS...tu non sei su HA OS. E comunque quando dai il comando docker per entrare nel container non ti appare quella scritta...riguarda bene :D

**Reply by MapoCaos** [reply]
Tu per avere quella finestra li sei passato per il terminal di HA? Io per riuscire ad everlo così vado sulla pagina homeassistant: 192.168.178.xxx:8123 e poi vado su Terminal che ho installato su HA tramite addons. Però c'è scritto che non funziona tramite addons o ssh... ho provato anche a inserire come ho visto nei tuoi screen: `ssh hassos` ma mi dice `ssh: Could not resolve hostname hassos: Name or service not known` (e cmq è sempre ssh!). Però non ho capito come hai fatto a raggiungere la console di HA sul tuo primo screen (che ha anche un nero differente ) sta cosa mi sfugge... l'unica volta che mi è comparsa è durante l'installazione di HA... mentre me mi si apre sempre come nel tuo terzo screen.. mi sento scarsissimo..

**Reply by snakuzzo** [reply]
Ti stai fissando su sta cosa!!! Che te frega di quella finestra dico io?!
Quella roba l'ho fatta su una VM di HA OS...la schermata che vedi è uno screenshot fatto da VirtualBox!
Ma poi cosa lanci i comandi ssh a caso?  `ssh hassos` lo posso lanciare io perché nella config del mio client ssh ho configurato l'host con il nome `hassos` con auth con chiave SSH.

**Reply by MapoCaos** [reply]
Mi fissavo su questa cosa perchè non sono ancora sicuro di aver capito quale sia la console di HA.. Avendo poca esperienza cerco di crearmi dei punti fermi in modo di avere delle certezze, solo per orientarmi. Tipo: "quando mi compare questa scritta sono nella console di HA". Però ho capito cosa intendi tu, ogni situazione è diversa quindi le cose possono cambiare, solo che devo avere più esperienza. Tra l'altro adesso che l'hai detto avevo usato anche io Virtualbox su Win10 con HA per delle prove ed era così. 
eh.. Per quanto riguarda i comandi, non tutti mi è chiaro cosa facciano.. per caso sapresti consigliarmi dove imparare qualcosa di più su debian e il suo funzionamento? qualcosa per dummies
Cmq grazie millle ancora per il supporto e il tempo che mi avete dedicato!

**Reply by snakuzzo** [reply]
Google
